### PR TITLE
proposal new design logic refactored

### PR DESCRIPTION
**Pull request by evidentti**

### CODE REVIEW

File: app/styles/proposal.scss [hunk 1]
```diff
@@ -11,10 +11,17 @@ h1, h2, h3, h4, h5 {
...
 
+span.glyphicon {
+    font-size: 1.0em !important;
+}
+
 .prop-element {
     background-color: white;
-    margin: 2px;
-    padding: 10px;
+    margin: 0.1em;
```
Comment (tkenakka) [inline app/styles/proposal.scss:20]: Not sure if instead of `em` responsive margins should use `vw` and `vh` ...

File: app/styles/proposal.scss [hunk 2]
```diff
@@ -69,14 +76,14 @@ h1, h2, h3, h4, h5 {
...
 .prop-label {
-    margin-top: 2px;
-    margin-bottom: 2px;
-    margin-left: 1px;
-    margin-right: 1px;
-    padding-top: 8px;
-    padding-bottom: 6px;
-    padding-left: 10px;
-    padding-right: 10px;
+    // margin-top: 2px;
```
Comment (tkenakka) [inline app/styles/proposal.scss:79]: Remove commented code?

File: test/karma.conf.js [hunk 3]
```diff
@@ -34,6 +34,7 @@ module.exports = function(config) {
       'bower_components/angular-translate/angular-translate.js',
       'bower_components/angular-translate-loader-static-files/angular-translate-loader-static-files.js',
       'bower_components/angular-translate-handler-log/angular-translate-handler-log.js',
+      'bower_components/angular-auto-focus/angular-auto-focus.js',
```
Comment (tkenakka) [inline test/karma.conf.js:37]: Isn't there a minified version?

File: app/index.html [hunk 4]
```diff
@@ -43,6 +43,7 @@
     <script src="bower_components/angular-translate/angular-translate.js"></script>
     <script src="bower_components/angular-translate-loader-static-files/angular-translate-loader-static-files.js"></script>
     <script src="bower_components/angular-translate-handler-log/angular-translate-handler-log.js"></script>
+    <script src="bower_components/angular-auto-focus/angular-auto-focus.js"></script>
```
Comment (tkenakka) [inline app/index.html:46]: Isn't there a minified version?

File: app/directives/proposalList/proposalList.directive.js [hunk 5]
```diff
@@ -187,15 +187,19 @@ angular.module('dashboard')
...
                 $log.debug("dbProposalList: getProposals: " + guid);
-                if (typeof guid === 'string') {
+                if (angular.isString(guid)) {
+                    self.proposals = [];
                     AhjoProposalsSrv.get({ 'guid': guid }).$promise.then(function (response) {
                         $log.debug("dbProposalList: get then");
                         if (angular.isObject(response) && angular.isArray(response.objects)) {
-                            self.proposals = angular.copy(response.objects);
+                            angular.forEach(response.objects, function (p) {
+                                if (angular.isObject(p) && angular.isString(p.proposalGuid)) {
```
Comment (tkenakka) [inline app/directives/proposalList/proposalList.directive.js:196]: Would checking for empty string be defensive programming or overkill?

**Comment by tkenakka** [comment]
###### On quick test
- triangle btn stacks to 2nd row
- Triangle button should not switch position between expanded/minimized states
- no feedback how long operation is ongoing: if it takes a while UI is not clear (progress info not yet implemented?)
- Cancelling creation of a new proposal doesn't delete the unsaved draft, use may think it's saved when it's not
- Confirmation dialog missing from  e.g. cancelling editing, but on 2nd thought maybe we go without it for now...

###### On mobile:
- On mobile only beginning of type displayed, content doesn't fit screen, maybe item  should display 2 rows
- When expanded proposal text overflows out of screen
- Publishing a proposal draft twice produces an exception about duplicate items

But maybe these are just not yet implemented

File: test/karma.conf.js (hunk 3 above)
Comment (evidentti) [inline test/karma.conf.js:37]: There is not a minified version.

File: app/index.html (hunk 4 above)
Comment (evidentti) [inline app/index.html:46]: There is not a minified version.

**Comment by tkenakka** [comment]
Ok to merge if mentioned issues will be improved in following PRs

**Pull request merged**